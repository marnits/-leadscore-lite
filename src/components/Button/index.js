import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.module.scss';

const Button = ({
  text,
  onClick,
  className,
  type,
}) => {
  const classes = `${styles.Button} ${styles[className]}`;

  return <button type={type} className={classes} onClick={onClick}>{text}</button>;
};

Button.defaultProps = {
  className: '',
  type: 'button',
};

Button.propTypes = {
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  className: PropTypes.string,
  type: PropTypes.string,
};

export default Button;
