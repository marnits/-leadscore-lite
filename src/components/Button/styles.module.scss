.Button {
  padding: 5px 16px;
  min-width: 60px;
  height: 30px;
  font-size: 13px;
  font-weight: bold;
  border-width: 0;
  border-radius: 6px;
  color: #33312e;
  cursor: pointer;
  outline: none;
  text-shadow: none;
}

.Button:hover:not(.disabled) {
  filter: brightness(97%);
}

.Button:active:not(.disabled) {
  box-shadow: inset 1px 1px 7px 0px rgba(84,84,84,0.2);
}

.Button.primary {
  background-image: linear-gradient(to bottom, #ffd532, #ffb500);
}

.Button.secondary {
  background-image: linear-gradient(to bottom, #e4dfdf, #c0c0c0);
}

.Button.disabled, .Button.primary.disabled, .Button.secondary.disabled {
  background-image: linear-gradient(to bottom, #f1efef, #dcdcdc);
  color: #b5b3b3;
  cursor: auto;
}
