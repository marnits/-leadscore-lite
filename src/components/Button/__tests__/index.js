import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Button from '../index';

describe('Button', () => {
  test('should render component properly', () => {
    const tree = renderer.create(<Button text="test" onClick={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render component properly', () => {
    const handleClick = jest.fn();
    const wrapper = shallow(
      <Button
        className="primary"
        text="test"
        onClick={handleClick}
        type="submit"
      />,
    );

    wrapper.find('button').simulate('click');

    expect(handleClick.mock.instances.length).toBe(1);
  });
});
