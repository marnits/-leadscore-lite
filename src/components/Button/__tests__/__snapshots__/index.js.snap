// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render component properly 1`] = `
<button
  className="Button "
  onClick={[Function]}
  type="button"
>
  test
</button>
`;

exports[`Label should render component properly 1`] = `
<button
  className="Button "
  type="button"
>
  label
</button>
`;

exports[`Label should render component properly 2`] = `
<button
  className="Button "
  type="button"
>
  label
</button>
`;
